e2e/testcases/basic_test.go

```diff
@@ -0,0 +1,365 @@
+
+package e2e
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/pkg/errors"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"kpt.dev/configsync/e2e/nomostest"
+	"kpt.dev/configsync/e2e/nomostest/ntopts"
+	nomostesting "kpt.dev/configsync/e2e/nomostest/testing"
+	v1 "kpt.dev/configsync/pkg/api/configmanagement/v1"
+	"kpt.dev/configsync/pkg/api/configsync"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	yamlDir = "../testdata"
+)
+
+// NamspaceConfigHasMetadataName will check the metadata name and compare it with the expected value
+func NamespaceConfigHasMetadataName(expectedName string) nomostest.Predicate {
+	return func(o client.Object) error {
+		raw := o.(*v1.NamespaceConfig).Spec.Resources[0].Versions[0].Objects[0].Raw
+		dep := &appsv1.Deployment{}
+		if err := json.Unmarshal(raw, &dep); err != nil {
+			return errors.Errorf("Failed to get raw JSON, %s", err)
+		}
+		actualName := dep.GetName()
+		if actualName != expectedName {
+			return errors.Errorf("Expected name: %s, got: %s", expectedName, actualName)
+		}
+		return nil
+	}
+}
+
+// NamspaceConfigHasMetadataCreationTimestamp will check the creation timestamp and compare it with the expected time
+func NamespaceConfigHasMetadataCreationTimestamp(expectedTime metav1.Time) nomostest.Predicate {
+	return func(o client.Object) error {
+		raw := o.(*v1.NamespaceConfig).Spec.Resources[0].Versions[0].Objects[0].Raw
+		dep := &appsv1.Deployment{}
+		if err := json.Unmarshal(raw, &dep); err != nil {
+			return errors.Errorf("Failed to get raw JSON, %s", err)
+		}
+		actualTime := dep.GetCreationTimestamp()
+		if actualTime != expectedTime {
+			return errors.Errorf("Expected time: %s, got: %s", expectedTime, actualTime)
+		}
+		return nil
+	}
+}
+
+func TestNoDefaultFieldsInNamespaceConfig(t *testing.T) {
+	nt := nomostest.New(t, nomostesting.Reconciliation2, ntopts.SkipMultiRepo)
+
+	nt.T.Log("Add a deployment")
+	nt.RootRepos[configsync.RootSyncName].Copy(fmt.Sprintf("%s/dir-namespace.yaml", yamlDir), "acme/namespaces/dir/namespace.yaml")
+	nt.RootRepos[configsync.RootSyncName].Copy(fmt.Sprintf("%s/deployment-helloworld.yaml", yamlDir), "acme/namespaces/dir/deployment.yaml")
+	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Adding deployment")
+	nt.WaitForRepoSyncs()
+
+	nt.T.Log("Check that the deployment was created")
+	if err := nt.Validate("hello-world", "dir", &appsv1.Deployment{}); err != nil {
+		nt.T.Fatal(err)
+	}
+
+	nt.T.Log("Check that specified field name is present")
+	if err := nt.Validate("dir", "", &v1.NamespaceConfig{}, NamespaceConfigHasMetadataName("hello-world")); err != nil {
+		nt.T.Fatal(err)
+	}
+
+	nt.T.Log("Check that unspecified field creationTimestamp is not present")
+	if err := nt.Validate("dir", "", &v1.NamespaceConfig{}, NamespaceConfigHasMetadataCreationTimestamp(metav1.Time{})); err != nil {
```

Comment:
Just to be clear, my recommendation was to move `NamespaceConfigHasMetadataName 
`/`NamespaceConfigHasMetadataCreationTimestamp` to be an anonymous/inline functions here (as opposed to moving the functions to this file).

IMO the function names are too generic sounding for what they are actually doing.

Reply:
sorry i misunderstood the original comments. Both functions are defined anonymously inline